src/js/components/Anchor.js

```diff
@@ -41,32 +46,27 @@ export default class Anchor extends Component {
 
     let classes = classnames(
       CLASS_ROOT,
-      this.props.className,
+      className,
       {
-        [`${CLASS_ROOT}--animate-icon`]: hasIcon &&
-          this.props.animateIcon !== false,
-        [`${CLASS_ROOT}--disabled`]: this.props.disabled,
-        [`${CLASS_ROOT}--icon`]: icon || hasIcon,
-        [`${CLASS_ROOT}--icon-label`]: hasIcon && this.props.label,
-        [`${CLASS_ROOT}--primary`]: this.props.primary,
-        [`${CLASS_ROOT}--reverse`]: this.props.reverse
+        [`${CLASS_ROOT}--animate-icon`]: hasIcon && animateIcon !== false,
+        [`${CLASS_ROOT}--disabled`]: disabled,
+        [`${CLASS_ROOT}--icon`]: anchorIcon || hasIcon,
+        [`${CLASS_ROOT}--icon-label`]: hasIcon && label,
+        [`${CLASS_ROOT}--primary`]: primary,
+        [`${CLASS_ROOT}--reverse`]: reverse
       }
     );
 
-    if (!children) {
-      children = this.props.label;
+    if (!anchorChildren) {
+      anchorChildren = label;
     }
 
-    const first = this.props.reverse ? children : icon;
-    const second = this.props.reverse ? icon : children;
+    const first = reverse ? anchorChildren : anchorIcon;
+    const second = reverse ? anchorIcon : anchorChildren;
 
-    const Component = this.props.tag;
+    const Component = tag;
     return (
-      <Component id={this.props.id} className={classes}
-        href={this.props.href}
-        target={this.props.target}
-        onClick={this.props.onClick}
-        aria-label={this.props.a11yTitle}>
+      <Component {...props} id={id} className={classes} aria-label={a11yTitle}>
```

Comment:
Or I can remove the assignment of `id` from line 15 & reference it on line 25 as `props.id`